### ISSUE TITLE
[Question] LRO version overriding?

### ISSUE THREAD

**Issue by HuairuoYang** [googleapis/gapic-generator-go#1193]
Hi Team! 
A quick question of why would you have this function:https://github.com/googleapis/gapic-generator-go/blob/761b041d6255da3c073d3267aeefcaac6cb991c1/internal/gengapic/mixins.go#L260

to override the LRO version since it's always `v1`?
https://chromium.googlesource.com/infra/goma/server/+/48224f31513604484d7c74afa5fd9d7e621e1691/proto/google/longrunning/operations.proto#55

**Comment by noahdietz** [comment]
_tl;dr: The base path is not always `/v1`, because each Google service exposes its own bindings for the Operations service._

Good question @HuairuoYang. This is getting a bit in the weeds of things, but all of the config is out there to tell this story.

So the [google/longrunning/operations.proto](https://github.com/googleapis/googleapis/blob/master/google/longrunning/operations.proto#L45-L54) is just an abstract interface that all Google Cloud Services implement themselves. This means they have their own HTTP bindings for the RPCs.  To copy the comments I linked to there:

> // When an API method normally takes long time to complete, it can be designed
// to return [Operation][google.longrunning.Operation] to the client, and the client can use this
// interface to receive the real response asynchronously by polling the
// operation resource, or pass the operation resource to another API (such as
// Google Cloud Pub/Sub API) to receive the response.  Any API service that
// returns long-running operations should implement the `Operations` interface
// so developers can have a consistent client experience.

The key phrase being:
> Any API service that returns long-running operations should implement the `Operations` interface

We refer to the Operations service as a "mixin" because it is a commonly defined surface that is "mixed in" to the API endpoint alongside its own service(s) ([more info here](https://google.aip.dev/client-libraries/4234)). This is captured in the `google.api.Service` config file fed to our generators ([example](https://github.com/googleapis/googleapis/blob/master/google/cloud/datastream/v1alpha1/datastream_v1alpha1.yaml#L42-L50)), but can also be derived from the proto package since the base path and version in the proto package are always the same.

**Comment by noahdietz** [comment]
If you have any more questions here, feel free to respond and I will reopen.
